com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -187,28 +187,50 @@ internal void RemoveOwnership(NetworkObject networkObject)
                 throw new SpawnStateException("Object is not spawned");
             }
 
-            for (int i = NetworkManager.ConnectedClients[networkObject.OwnerClientId].OwnedObjects.Count - 1;
-                i > -1;
-                i--)
+            // If we made it here then we are the server and if the server is determined to already be the owner
+            // then ignore the RemoveOwnership invocation.
+            if (networkObject.OwnerClientId == NetworkManager.ServerClientId)
             {
-                if (NetworkManager.ConnectedClients[networkObject.OwnerClientId].OwnedObjects[i] == networkObject)
+                if (NetworkLog.CurrentLogLevel >= LogLevel.Normal)
                 {
-                    NetworkManager.ConnectedClients[networkObject.OwnerClientId].OwnedObjects.RemoveAt(i);
+                    NetworkLog.LogWarning($"Remove ownership ignored:  {networkObject.name} is already set to the default server ownership!");
```

Comment:
I wonder if this is actually necessary, couldn't we just call this idempotent? "You were trying to remove ownership. It was already owned by server. Nothing to do here"
It seems to me that this warning would just be annoying to devs, because they would have to add unnecessary checks around their code to prevent it

Reply:
Yeah, I originally made it silently return but then figured if you are trying to remove ownership immediately after you spawn something (i.e. the next line of code after spawning) then you would get this exception.  This was how the bug was found, and as such I opted to leave the message there as it only will display that message if you spawn a NetworkObject on the server and then immediately try to remove ownership...so if a user does this they know they don't need to remove any ownership at that point since it is already spawned with the default server ownership.

Good call out as this was what I deliberated over the most.  Even though it won't spam (unless a user is trying to do something they don't have to/need to), does that change your thoughts on logging this warning?

Reply:
Not really - I mean, what are we warning the user about? That they didn't need to call the function? Every piece of information we tell the user has a cost, and since the consequence here is entirely innocuous, I still think it's not in the users best interest to throw this up because there's not really an actual problem here and we've created noise in their logs. Unless I'm misunderstanding here.

I also think there may be additional cases beyond what you described. What about users that are cleaning up an object's state (say, from a disconnected client or a client that no longer is allowed ownership of this object by the game mechanics) and the user's code defensively calls this multiple times? Maybe they don't _need_ to, but there's no actual consequence to them doing so.

That said, I would leave the decision in your hands. It's not a big sticking point for me.

Reply:
what about `if (NetworkLog.CurrentLogLevel == LogLevel.Developer)`? :)
(I think `NetworkLog.CurrentLogLevel >= LogLevel.Normal` is a bug, you probably wanted to make it `<=` in the first place)

Reply:
but FWIW, +1 on @becksebenius-unity's point, hence why I suggested a middleground with lesser important LogLevel.